From: Damian Hobson-Garcia <dhobsong@igel.co.jp>
Date: Thu, 28 Sep 2017 18:20:42 +0900
Subject: [PATCH] appsrc: Clear EOS on flush

The currently preferred method of flushing a pipeline is to use
the FLUSH_START/STOP events, but these events are input at the
source pad of the pipeline source, and therefore are not passed
to the source plugin callback.  Instead do the processing in the
event callback on the plugin source pad.
---
 gst-libs/gst/app/gstappsrc.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/gst-libs/gst/app/gstappsrc.c b/gst-libs/gst/app/gstappsrc.c
index fff99e9..1bbe7b1 100644
--- a/gst-libs/gst/app/gstappsrc.c
+++ b/gst-libs/gst/app/gstappsrc.c
@@ -779,6 +779,7 @@ gst_app_src_send_event (GstElement * element, GstEvent * event)
     case GST_EVENT_FLUSH_STOP:
       g_mutex_lock (&priv->mutex);
       gst_app_src_flush_queued (appsrc, TRUE);
+      priv->is_eos = FALSE;
       g_mutex_unlock (&priv->mutex);
       break;
     default:
--
1.9.1

